### PR TITLE
8342707: Prepare Gatherers for graduation from Preview

### PR DESCRIPTION
**Pull request by viktorklang-ora**
Make final adjustments to drop PreviewFeature and updating the @ since markers.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Change requires CSR request [JDK-8343323](https://bugs.openjdk.org/browse/JDK-8343323) to be approved
- [x] Commit message must refer to an issue

### Issues
 * [JDK-8342707](https://bugs.openjdk.org/browse/JDK-8342707): Prepare Gatherers for graduation from Preview (**Enhancement** - P4)
 * [JDK-8343323](https://bugs.openjdk.org/browse/JDK-8343323): Prepare Gatherers for graduation from Preview (**CSR**)

### Reviewers
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)
 * [Chen Liang](https://openjdk.org/census#liach) (@liach - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21686/head:pull/21686` \
`$ git checkout pull/21686`

Update a local copy of the PR: \
`$ git checkout pull/21686` \
`$ git pull https://git.openjdk.org/jdk.git pull/21686/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21686`

View PR using the GUI difftool: \
`$ git pr show -t 21686`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21686.diff">https://git.openjdk.org/jdk/pull/21686.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21686#issuecomment-2435575169)
</details>

### CODE REVIEW

File: src/java.base/share/classes/jdk/internal/javac/PreviewFeature.java [hunk 1]
```diff
@@ -76,7 +76,6 @@ public enum Feature {
         STRUCTURED_CONCURRENCY,
         @JEP(number=466, title="ClassFile API", status="Second Preview")
         CLASSFILE_API,
-        @JEP(number=473, title="Stream Gatherers", status="Second Preview")
         STREAM_GATHERERS,
```
Comment (liach) [inline src/java.base/share/classes/jdk/internal/javac/PreviewFeature.java:79]: Does bootcycle fail if this enum constant is removed?

Comment (viktorklang-ora) [reply src/java.base/share/classes/jdk/internal/javac/PreviewFeature.java:79]: Yep :)

**Comment by AlanBateman** [comment]
Looks fine, you'll need to sync up the PR to resolve the merge conflict, and you'll need to update the copyright header of several files (esp. tests) before integrate.

**Comment by viktorklang-ora** [comment]
@liach @AlanBateman rebase on current mainline and fixed the conflicts and pushed the update. The CSR is accepted so this should be ready to merge.

**Review by AlanBateman — APPROVED** [review]

**Review by liach — APPROVED** [review]
The removal of preview toggles look good. Confirmed that since in stream packages and removal of preview toggles tests/micro look good.

Side comment: With the `jvmArgs` cleared on benchmark `@Fork`, you can just make it `@Fork(1)` instead of `@Fork(value = 1)`.

**Comment by viktorklang-ora** [comment]
@liach 

>Side comment: With the jvmArgs cleared on benchmark @Fork, you can just make it @Fork(1) instead of @Fork(value = 1).

Yeah, I opted to keep it as-is to make the delta between old-and-new low (and to make adding additional params slightly faster). I'd be happy to change it if you insist, tho! :)

**Comment by liach** [comment]
The current form makes it easy to add jvm args in the future, too.  It's totally fine.

**Pull request closed without merging**